tools/sanity_checks.py

```diff
@@ -72,6 +72,21 @@ def test_releases_json(self):
             for k in info.keys():
                 self.assertIn(k, PERMITTED_KEYS)
 
+    def get_patch_path(self, wrap_section):
+        patch_directory = wrap_section.get('patch_directory')
+        if patch_directory:
+            return Path('subprojects', 'packagefiles', patch_directory)
+
+        return None
+
+    def check_meson_version(self, name: str, version: str,patch_path: str, builddir: str = '_build'):
+        with self.subTest(step="check_meson_version"):
+            with open(Path(builddir) / "meson-info/intro-projectinfo.json") as project_info_file:
+                project_info = json.load(project_info_file)
+                subproject, = [subproj for subproj in project_info["subprojects"] if subproj["name"] == name]
+                if subproject['version'] != 'undefined' and not patch_path:
```

Comment:
So we only run the check if the patch_path does not exist -- i.e. meson.build comes from upstream? Oops.